### DISCUSSION TITLE
Trying to use hover-effect in a functional component

### DISCUSSION THREAD

**Discussion by fredcorr** [vercel/next.js#15446] [Help] [answered]
Hi
I'm trying to use an npm package called [hover-effect](https://www.npmjs.com/package/hover-effect) inside a functional component, but I'm constantly bumping into some issues. I have used similar packages in create-react-app before but I'm new to next.js. I would appreciate some help, and sorry if this issue seems very easy. See below how the package should be used:

```
import { fade, scaleUp, slideX } from '../../util/animation';
import { useInView } from 'react-intersection-observer'
import React, {useRef,useEffect} from "react";
import styles from './HomePannel.module.css';
import hoverEffect from 'hover-effect';
import { motion } from 'framer-motion';
import Link from 'next/link';

const homePannel = (props) => {

  const [ref, inView ] = useInView({ threshold: 0.7, triggerOnce: true })

  const container = useRef(null)

  useEffect(()=> {
    new hoverEffect({
      parent: container,
      image1: props.coverImg,
      image2:props.coverImg,
      displacementImage: props.coverImg
    })
   }
  ,[])

  return (
    <Link href="/case-studies/[case-study]" as={"/case-studies/" + props.link } scroll={false}>
      <a className={ styles.thumb } key={ props._key } ref={ ref }>
        <HoverEffect />
        <div className={ styles.cover } >
          <motion.p
            animate={inView ? "show" : "hidden"}
            initial="hidden"
            variants={fade}
            alt="">
              { props.details }
          </motion.p>
          <motion.div 
            style={{ backgroundImage: "url(" + props.coverImg + ""}}
            animate={inView ? "show" : "hidden"}
            ref={ container }
            variants={scaleUp}
            initial="hidden"
            exit="hidden"
            >
          </motion.div>
          <motion.p  
            animate={inView ? "show" : "hidden"}
            initial="hidden"
            variants={slideX( 100 )}
            exit="hidden"
            custom={ 100, 0.1 }>
              { props.date }
          </motion.p>
        </div>
        <motion.h2 
          animate={inView ? "show" : "hidden"}
          initial="hidden"
          variants={slideX( 100 )}
          custom={ 100, 0.4 }
          exit="hidden">
            { props.title }
        </motion.h2>
      </a>
    </Link>
  )
}

export default homePannel;
```

First of all, I tried to import it like any other npm module, but I get the issue
`SyntaxError: Cannot use import statement outside a module`

I have tried to import the package dynamically using next/dynamic in this way:

`const HoverEffect = dynamic(() => import('hover-effect').then((a) => a.hover-effect), {ssr: false});`

But when I try to use 'new' syntax I get the error 'HoverEffect' is not a constructor. 

Anyone aware of any solution would be a lifesaver
Best

**Comment by jamesmosier** [comment, marked as answer]
It looks like the issue is that `hover-effect` imports a module that is not transpiled which is `gsap`. If you look at `node_modules/hover-effect/dist/hover-effect.js` you'll see `gsap/TweenMax` is required and then inside that module they use the `import` syntax which is not supported by Next, hence why it needs transpiled.

As a workaround, you could use [next-transpile-modules](https://github.com/martpie/next-transpile-modules) and transpile the `gsap` module. 

```js
// next.config.js
const withTM = require('next-transpile-modules')(['gsap']);
module.exports = withTM();
```

**Reply by alpharaoh** [reply]
I have tried to use transpile but I still run into the same error

**Reply by justincapp** [reply]
> I have tried to use transpile but I still run into the same error

Did you resolve this ?
